src/server/routes/redteam.ts

```diff
@@ -2,120 +2,130 @@ import { Router } from 'express';
 import { v4 as uuidv4 } from 'uuid';
 import cliState from '../../cliState';
 import logger from '../../logger';
-import { REDTEAM_MODEL } from '../../redteam/constants';
-import { Plugins } from '../../redteam/plugins/index';
+import {
+  REDTEAM_MODEL,
+  ALL_PLUGINS,
+  ALL_STRATEGIES,
+  isMultiTurnStrategy,
+  type MultiTurnStrategy,
+  type Plugin,
+  type Strategy,
+} from '../../redteam/constants';
+import { PluginFactory, Plugins } from '../../redteam/plugins/index';
 import { redteamProviderManager } from '../../redteam/providers/shared';
 import { getRemoteGenerationUrl } from '../../redteam/remoteGeneration';
 import { doRedteamRun } from '../../redteam/shared';
+import { Strategies } from '../../redteam/strategies/index';
 import { fetchWithProxy } from '../../util/fetch/index';
 import { evalJobs } from './eval';
 import type { Request, Response } from 'express';
+import { z } from 'zod';
+import {
+  ConversationMessageSchema,
+  PluginConfigSchema,
+  StrategyConfigSchema,
+} from '../../redteam/types';
+import { type Strategy as StrategyFactory } from '../../redteam/strategies/types';
+import {
+  extractGeneratedPrompt,
+  generateMultiTurnPrompt,
+  getPluginConfigurationError,
+  RemoteGenerationDisabledError,
+} from '../services/redteamTestCaseGenerationService';
 
 export const redteamRouter = Router();
 
-// Generate a single test case for a specific plugin
-redteamRouter.post('/generate-test', async (req: Request, res: Response): Promise<void> => {
-  try {
-    const { pluginId, config } = req.body;
-
-    if (!pluginId) {
-      res.status(400).json({ error: 'Plugin ID is required' });
-      return;
-    }
-
-    // Find the plugin
-    const plugin = Plugins.find((p) => p.key === pluginId);
-    if (!plugin) {
-      res.status(400).json({ error: `Plugin ${pluginId} not found` });
-      return;
-    }
+function respondWithMultiTurnError(error: unknown, res: Response, strategyId: Strategy) {
+  if (error instanceof RemoteGenerationDisabledError) {
+    res.status(400).json({ error: error.message });
```

Comment:
Could have another function that wraps `respondWithMultiTurnError` and does this RemoteGenerationDisabledError check there, instead of shoehorning it in here. Or could just rename this function.